### ISSUE TITLE
[YSQL] Handle scanning >1024 rows per tablet during backfill

### ISSUE THREAD

**Issue by jaki** [yugabyte/yugabyte-db#5031]
During the backfill stage of multi-stage index create, it first reads rows from the indexed table then writes rows to the index table.  The reads are done in parallel to each of the tablets of the indexed table.  The problem is when a tablet has more than 1024 rows, which comes from the flag `ysql_prefetch_limit`, the scan returns, and we're supposed to send another request with paging state saying what next row key to read.  However, that next request isn't getting sent.

Either make the limit higher (unlimited?) for backfill purposes or handle the paging state to send further read requests.

Repro:

```sh
./bin/yb-ctl create --master_flags "ysql_disable_index_backfill=false" --tserver_flags "ysql_disable_index_backfill=false" --rf 1 --ysql_num_shards_per_tserver 1
./bin/ysqlsh
```

```sql
CREATE TABLE big (i int);
INSERT INTO big VALUES (generate_series(1, 1025));
CREATE INDEX ON big (i ASC);
EXPLAIN SELECT COUNT(*) FROM big WHERE i > 0; -- index only scan
SELECT COUNT(*) FROM big WHERE i > 0;
```

Expect (good):

```
 count 
-------
  1025
(1 row)
```

Actual (bad):

```
 count 
-------
  1024
(1 row)
```

**Comment by jaki** [comment]
This causes `TestPgRegressTypesUDT` java test to fail when backfill is enabled.

**Comment by jaki** [comment]
The solution for this is dependent on handling range tables for `COUNT(*)`: see issue #4768.

**Comment by jaki** [comment]
Let's leave handling backfilling range partitioned tables to >1 tablets as a separate task.  So this issue tracks getting this working for all hash partitioned tables and range partitioned tables of a single tablet.
